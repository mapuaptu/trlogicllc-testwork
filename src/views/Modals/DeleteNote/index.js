import Button from '@/components/Button/index.vue';

export default {
  name: 'ModalDeleteNote',
  components: {
    Button,
  },
  props: {
    id: {
      type: Number,
      default: null,
    },
    name: {
      type: String,
      default: '',
    },
  },
  methods: {
    onCancel() {
      return this.$emit('close');
    },
    onDelete() {
      this.$store.commit('DELETE_NOTE', { id: this.id });

      return this.$emit('close');
    },
  },
};
