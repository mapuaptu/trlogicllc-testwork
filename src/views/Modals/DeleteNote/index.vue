<template>
  <div
    :class="$style.modal"
    class="modal"
  >
    <div class="modal-header">
      Delete Note
    </div>

    <div class="modal-content">
      Are you sure you want to delete note {{ name }}
    </div>

    <div class="modal-footer">
      <Button
        :fill="true"
        @click.native="onCancel"
      >
        Cancel
      </Button>

      <Button @click.native="onDelete">
        Delete
      </Button>
    </div>
  </div>
</template>

<script src="./index.js"></script>
<style src="./index.scss" lang="scss" module></style>
