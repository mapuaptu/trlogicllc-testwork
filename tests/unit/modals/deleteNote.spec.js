import DeleteNote from '@/views/Modals/DeleteNote/index.vue';
import { mount } from '@vue/test-utils';

describe('Modal createNote component', () => {
  it('1. Rendered success', async () => {
    const wrapper = await mount(DeleteNote);

    expect(wrapper.html()).toMatchSnapshot();
  });
});
