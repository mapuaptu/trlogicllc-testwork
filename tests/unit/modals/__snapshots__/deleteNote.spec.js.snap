// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal createNote component 1. Rendered success 1`] = `
<div class="modal">
  <div class="modal-header">
    Delete Note
  </div>
  <div class="modal-content">
    Are you sure you want to delete note
  </div>
  <div class="modal-footer">
    <div class=""><span>
      Cancel
    </span>
      <!---->
    </div>
    <div class=""><span>
      Delete
    </span>
      <!---->
    </div>
  </div>
</div>
`;
